src/course/put-course.js

```diff
@@ -0,0 +1,58 @@
+const dynamodb = require('aws-sdk/clients/dynamodb');
+const db = new dynamodb.DocumentClient();
+const { statusCode, inputFormat, reponseFormat, errFormat } = require('../util')
+// Get the DynamoDB table name from environment variables
+const tableName = process.env.TABLE;
+const crypto = require('crypto')
+
+/**
+ * A simple example includes a HTTP post method to add one item to a DynamoDB table.
+ */
+exports.putCourseHandler = async (event, context, callback) => {
+    const { body, httpMethod, path } = event;
+    let response = null;
+
+    // Get id and name from the body of the request
+    const {
+    CourseName,
+    CourseID,
+    Description,
+    Syllabus,
+    Session,
+    ID,
+    } = typeof body === 'string' ? JSON.parse(body) : body;
+
+    //Checking for errors
+    if (httpMethod !== 'POST') {
+        return errFormat(statusCode.MethodNotAllow, `postMethod only accepts POST method, you tried: ${httpMethod} method.`, callback)
+    }
+    else if (!CourseID || !CourseName) {
+        return errFormat(statusCode.BadRequest, `Missing CourseID or CourseName.`, callback)
+    }
+
+    if (!response) {
+        //combine them into an object
+        const req = {
+            CourseName,
+            CourseID,
+            Description,
+            Syllabus,
+            Session,
+            ID : ID ? ID : 
+            crypto.createHash('sha1').update(CourseName+CourseID+Session+Discription+Syllabus).digest('hex')
```

Comment:
Miss-spell : Description

Generally it's a bad idea to hash the entire Description and the Syllabus because they are heavy. You just need someway to get unique information. maybe adding date and time would be good.

Reply:
no need for the entire description and syllabus

Reply:
Can I just simply hash the CourseID only?

Reply:
yes